yt/frontends/open_pmd/data_structures.py

```diff
@@ -96,8 +96,8 @@ def _get_particle_type_counts(self):
                     spec = list(f[bp + pp].keys())[0]
                 else:
                     spec = ptype
-                axis = list(f[bp + pp + "/" + spec + "/position"].keys())[0]
-                pos = f[bp + pp + "/" + spec + "/position/" + axis]
+                axis = list(f[os.path.join(bp + pp, spec, "position")].keys())[0]
+                pos = f[os.path.join(bp + pp, spec, "position", axis)]
```

Comment:
Shouldn't it be the following instead?
```suggestion
                pos = f[os.path.join(bp, pp, spec, "position", axis)]
```

Reply:
Good catch. Indeed `self.base_path` is also currently hardcoded in a POSIX-only fashion, I'll fix that too

Reply:
actually I'm not sure what to do here. These path components are also used as keys for `h5py.File` objects, and I can't find a clear reference for how these behave on windows. I think `"/"` should still work there so I don't actually need to edit this particular frontend.